webapp/sources/rudder/rudder-rest/src/main/scala/com/normation/rudder/users/RudderUserDetailsFile.scala

```diff
@@ -76,34 +75,18 @@ final case class UserFile(
     inputStream: () => InputStream
 )
 
+sealed abstract class PasswordEncoder(override val entryName: String) extends EnumEntry {
+  def name:        String = entryName
+  def displayName: String = name
+  def encoder:     PasswordEncoder.Rudder
+}
```

Comment:
We need this ADT because BCRYPT specifically has 1 additional parameter (`cost: Int`) which is a configuration value used to implement the `encoder`. It also allows easier parsing since it is an enum from _enumeratum_